tests/pc/test_po_update.py

```diff
@@ -111,8 +132,132 @@ def test_po_update(duthosts, enum_rand_one_per_hwsku_frontend_hostname, enum_fro
+        asichost.config_portchannel_member(pc, member, "del")
+
+    # Step 2: Remove port channel ip from port channel
+    asichost.config_ip_intf(pc, pc_ip + "/31", "remove")
+
+    time.sleep(30)
+    int_facts = asichost.interface_facts()['ansible_facts']
+    pytest_assert(not int_facts['ansible_interface_facts'][pc]['link'])
+    pytest_assert(wait_until(120, 10, 0, asichost.check_bgp_statistic, 'ipv4_idle', 1))
+
+    # Step 3: Create tmp port channel with default min-links(1)
+    asichost.config_portchannel(tmp_pc, "add")
+
+    # Step 4: Add port channel members to tmp port channel
+    for member in pc_members:
+        asichost.config_portchannel_member(tmp_pc, member, "add")
+
+    # Step 5: Add port channel ip to tmp port channel
+    asichost.config_ip_intf(tmp_pc, pc_ip + "/31", "add")
+    int_facts = asichost.interface_facts()['ansible_facts']
+    pytest_assert(int_facts['ansible_interface_facts'][tmp_pc]['ipv4']['address'] == pc_ip)
+
+    time.sleep(30)
+    int_facts = asichost.interface_facts()['ansible_facts']
+    pytest_assert(int_facts['ansible_interface_facts'][tmp_pc]['link'])
+    pytest_assert(wait_until(120, 10, 0, asichost.check_bgp_statistic, 'ipv4_idle', 0))
+
+    # Keep sending packets, and add/del different members during that time, observe whether packets lose
+    pkt = testutils.simple_ip_packet(
+        eth_dst=duthost.facts["router_mac"],
+        eth_src=ptfadapter.dataplane.get_mac(0, in_ptf_index),
+        ip_src=in_peer_ip,
+        ip_dst=out_peer_ip)
+
+    exp_pkt = pkt.copy()
+    exp_pkt = mask.Mask(exp_pkt)
+
+    exp_pkt.set_do_not_care_scapy(packet.Ether, 'dst')
+    exp_pkt.set_do_not_care_scapy(packet.Ether, 'src')
+    exp_pkt.set_do_not_care_scapy(packet.IP, 'chksum')
+    exp_pkt.set_do_not_care_scapy(packet.IP, 'ttl')
+
+    ptfadapter.dataplane.flush()
+    q = Queue(1)
+
+    def del_add_members():
+        time.sleep(1)
+        asichost.config_portchannel_member(tmp_pc, pc_members[0], "del")
+        time.sleep(2)
+        asichost.config_portchannel_member(tmp_pc, pc_members[0], "add")
+        time.sleep(2)
+        asichost.config_portchannel_member(tmp_pc, pc_members[1], "del")
+        time.sleep(2)
+        asichost.config_portchannel_member(tmp_pc, pc_members[1], "add")
+        time.sleep(5)
+        q.put(True)
+
+    t = threading.Thread(target=del_add_members, name="del_add_members_thread")
+    t.start()
+    t_max = time.time() + 60
+    send_count = 0
+
+    stop_sending = False
+    while not stop_sending:
+        testutils.send(ptfadapter, in_ptf_index, pkt)
+        send_count += 1
+        member_update_thread_finished = (not q.empty()) and q.get()
+        reach_max_time = time.time() > t_max
+        stop_sending = reach_max_time or member_update_thread_finished
+    t.join()
+
+    match_cnt = testutils.count_matched_packets_all_ports(ptfadapter, exp_pkt, ports=out_ptf_indices)
+
+    assert match_cnt > 0
+    assert match_cnt == send_count
```

Comment:
Please use pytest_assert as rest of the file and give a proper error message.

Reply:
Refined.